main.go

```diff
@@ -96,24 +104,22 @@ func lastDayWeek(lastDay time.Time) time.Time {
 // Function to get project ID from the gitLabAPI
 func getProjectID(baseURL string, token string, projectname string, namespace string) (string, error) {
 	projects := []gitLabAPI{}
-
 	client := &http.Client{}
-	req, err := http.NewRequest("GET", baseURL, nil)
+	strURL := []string{baseURL, "/projects/"}
+	URL := strings.Join(strURL, "")
+	req, err := http.NewRequest("GET", URL, nil)
```

Comment:
A better way to this would be just appending two strings with + operator since you only have 2 strings. Because using strings.Join and building a slice takes more time and uses more memory. Check this blog post for more information: http://herman.asia/efficient-string-concatenation-in-go
```
URL := baseURL + "/projects/"
```

Reply:
Question is are we really resource constrained and what is the more readable version. Especially with more complicated concatenations. I agree that both can be used and there are pros and cons for each way. Just wanted to note that readability might be favored. In addition to consistent behavior with more complicated concatenations.
So my vote is to leave it, but definitely remember the resource "burden" we are taking here.